webapp/channels/src/components/common/hooks/usePost.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type {Post} from '@mattermost/types/posts';
+
+import {getPostsByIdsBatched} from 'mattermost-redux/actions/posts';
+import {getPost} from 'mattermost-redux/selectors/entities/posts';
+
+import {makeUseEntity} from './useEntity';
+
+export const usePost = makeUseEntity<Post>({
```

Comment:
At this pace we may end up reimplementing apollo XD

Reply:
That's definitely possible, but our version will be backwards compatible with our existing Redux stores. I've already got ideas to add refetching options, WebSocket subscriptions for real-time syncing, and staleness detection for non-real-time syncing too 😀